tools/instrument_flag_consistency_check.php

```diff
@@ -0,0 +1,130 @@
+<?php
+    function pad_or_trunc ($str, $len) {
+        if (strlen($str) > $len) {
+            return substr($str, 0, $len-1) . ".";
+        } else if (strlen($str) == $len) {
+            return $str;
+        } else {
+            return str_pad($str, $len);
+        }
+    }
+    
+    require_once __DIR__ . "/../vendor/autoload.php";
+    $client = new NDB_Client();
+    $client->makeCommandLine();
+    $client->initialize(__DIR__ . "/../project/config.xml");
+    $DB = Database::singleton();
+    
+    $instrument_arr = $DB->pselect("
+        SELECT
+            Test_name
+        FROM
+            test_names
+    ", array());
+    
+    $check_arr = array();
+    echo "\n== Checking if instrument tables exist ==\n";
+    foreach ($instrument_arr as $instrument) {
+        $name = $instrument["Test_name"];
+        $exists = $DB->tableExists($name);
+        echo ($exists ?
+            "(true)  - " :
+            "(false) - "
+        );
+        echo "{$name}\n";
+        
+        if ($exists) {
+            $check_arr[] = $name;
+        }
+    }
+    
+    echo "\n== Checking for orphaned instrument/flag ==\n";
+    foreach ($check_arr as $check) {
+        echo "Checking {$check}...\n";
+        
+        $orphan_flag_arr = $DB->pselect("
+            SELECT
+                flag.*
+            FROM
+                flag
+            LEFT JOIN
+                {$check}
+            ON
+                {$check}.CommentID = flag.CommentID
+            WHERE
+                Test_name = :check AND
+                {$check}.CommentID IS NULL
+        ", array(
+            "check" => $check
+        ));
+        
+        $flag_str = pad_or_trunc($check, 5) . " flag";
+        foreach ($orphan_flag_arr as $orphan_flag) {
+            $comment_id = $orphan_flag["CommentID"];
+            echo "Warning, orphan {$flag_str}:{$comment_id}\n";

```

Comment:
@AnyhowStep make a distinction between flag orphans and instrument orphans (line 82)